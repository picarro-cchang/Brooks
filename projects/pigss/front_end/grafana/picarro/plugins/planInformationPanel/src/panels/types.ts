import { any } from "prop-types";

export interface Plan {
  max_steps: number;
  panel_to_show: number;
  current_step: number;
  focus: PlanFocus;
  last_step: number;
  steps: { [key: string]: PlanStep };
  num_plan_files: number;
  plan_files: { [key: string]: string };
  plan_filename: string;
  bank_names: {
    [key: number]: {
      name: string;
      channels: { [key: number]: string };
    };
  };
}

export interface PlanFocus {
  row: number;
  column: number;
}

export interface BankConfig {
  chan_mask: number;
  clean: number;
}

export interface PlanStep {
  banks: { [key: number]: BankConfig };
  reference: number;
  duration: number;
}
