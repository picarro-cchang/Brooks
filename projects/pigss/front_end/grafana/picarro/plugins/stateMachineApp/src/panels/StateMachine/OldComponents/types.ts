// import { any } from "prop-types";

// export interface ButtonInfo {
//   caption: string;
//   className: string;
//   response: string;
// }

// export interface CommandPanelOptions {
//   uistatus: {
//     [key: string]: string;
//   };
//   ws_sender: (o: object) => void;
//   plan: Plan;
//   updatePanel: (x: number) => void;
//   layoutSwitch: () => {}
// }

// export interface ModalInfo {
//   show: boolean;
//   html: string;
//   num_buttons: number;
//   buttons: { [key: string]: ButtonInfo };
// }

// export interface Plan {
//   max_steps: number;
//   panel_to_show: number;
//   current_step: number;
//   focus: PlanFocus;
//   last_step: number;
//   steps: { [key: string]: PlanStep };
//   num_plan_files: number;
//   plan_files: { [key: string]: string };
//   plan_filename: string;
//   bank_names: {
//     [key: number]: {
//       name: string;
//       channels: { [key: number]: string };
//     };
//   };
// }

// export interface PlanFocus {
//   row: number;
//   column: number;
// }

// export interface PlanLoadPanelOptions {
//   plan: Plan;
//   ws_sender: (o: object) => void;
//   isChanged: boolean;
//   updateFileName: (x: boolean) => void;
//   updatePanel: (x: number) => void;
//   loadPlan: (x: string) => void;
//   deleteFile: (x: string) => void;
//   getPlanFileNames: () => void;
//   fileNames: {[key: string]: string}
//   cancelLoadPlan: () => void;
// }

// export interface PlanLoadPanelEditOptions {
//   plan: Plan;
//   ws_sender: (o: object) => void;
//   isChanged: boolean;
//   updateFileName: (x: boolean) => void;
//   getPlanFromFileName: (s: string) => void;
//   editPlan: () => void;
//   updatePanel: (x: number) => void;
//   getPlanFileNames: () => void;
//   fileNames: {[key: string]: string}
// }

// export interface PlanPanelOptions {
//   uistatus: { [key: string]: string };
//   plan: Plan;
//   setFocus: (row: number, column: number) => void;
//   ws_sender: (o: object) => void;
//   isChanged: boolean;
//   updateFileName: (x: boolean) => void;
//   // addChanneltoPlan: (bank: number, channel: number) => void;
//   bankAddition: { [key: string]: number };
//   fileName: string;
//   loadFile: () => void;
//   savePlan: () => void;
//   updatePanel: (x: number) => void;
// }

// export interface PlanPanelLayoutOptions {
//   uistatus: { [key: string]: string };
//   plan: Plan;
//   setFocus: (row: number, column: number) => void;
//   ws_sender: (o: object) => void;
//   isChanged: boolean;
//   updateFileName: (x: boolean) => void;
//   // addChanneltoPlan: (bank: number, channel: number) => void;
//   bankAddition: { [key: string]: number };
//   updatePanel: (x: number) => void;            
//   getPlanFileNames: () => void;
//   fileNames: {[key: string]: string}
// }

// export enum PlanPanelTypes {
//   NONE = 0,
//   PLAN = 1,
//   LOAD = 2,
//   EDIT = 3,
//   PREVIEW = 4
// }

// export interface PlanSavePanelOptions {
//   plan: Plan;
//   ws_sender: (o: object) => void;
//   isChanged: boolean;
//   updateFileName: (x: boolean) => void;
//   planSaved: (f: string) => void;
//   editPlan: () => void;
//   updatePanel: (x: number) => void;
//   fileNames: {[key: string]: string}
// }

// export interface BankConfig {
//   chan_mask: number;
//   clean: number;
// }

// export interface PlanStep {
//   banks: { [key: number]: BankConfig };
//   reference: number;
//   duration: number;
// }

// export interface EditPanelOptions {
//   uistatus: {
//     [key: string]: any;
//   };
//   ws_sender: (o: object) => void;
//   plan: Plan;
//   updatePanel: (x: number) => void;

// }
