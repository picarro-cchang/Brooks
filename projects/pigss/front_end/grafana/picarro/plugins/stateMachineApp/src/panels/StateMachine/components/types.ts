export interface PlanFocus {
    row: number;
    column: number;
}

export interface BankConfig {
    chan_mask: number;
    clean: number;
}

export interface PlanStep {
    banks: { [key: number]: BankConfig };
    reference: number;
    duration: number;
}

export interface Plan {
    max_steps: number;
    panel_to_show: number;
    current_step: number;
    focus: PlanFocus;
    last_step: number;
    steps: { [key: string]: PlanStep };
    num_plan_files: number;
    plan_files: { [key: string]: string };
    plan_filename: string;
    bank_names: {
      [key: number]: {
        name: string;
        channels: { [key: number]: string };
      };
    };
}

export interface PlanLayoutProps {
    uistatus: {
      [key: string]: any;
    };
    // ws_sender: (o: object) => void;
    plan: Plan;
    layoutSwitch: () => void;
    fileNames: {}
    getPlanFileNames: () => void;
    loadedFileName: string;
}

export interface RunLayoutProps {
    uistatus: {
      [key: string]: any;
    };
    ws_sender: (o: object) => void;
    plan: Plan;
    layoutSwitch: () => void;
    fileNames: {}
    loadedFileName: string;
    getPlanFileNames: () => void;
}

export enum PanelTypes {
    NONE = 0,
    LOAD = 1,
    EDIT = 2,
    PREVIEW = 3
}

export enum PlanPanelTypes {
    PLAN = 0,
    LOAD = 1,
    SAVE = 2,
}

export interface BankPanelPlanOptions {
    bank: number;
    uistatus: {
      bank?: { [bankNum: string]: string };
      clean?: { [bankNum: string]: string };
      channel?: { [bankNum: string]: { [channelNum: string]: string } };
    };
    // ws_sender: (o: object) => void;
    plan: Plan;
    addChanneltoPlan: (bank: number, channel: number) => void;
}

export interface LoadPanelCommandOptions {
    plan: Plan;
    ws_sender: (o: object) => void;
    updatePanel: (x: number) => void;
    // loadPlan: (x: string) => void;
    // deleteFile: (x: string) => void;
    fileNames: {}
    cancelLoadPlan: () => void;
    getPlanFromFileName: (fileName: string) => void;
    deleteFile: (file: string) => void;
    loadedFileName: string;
    planID: String;
  }

  export interface PreviewPanelOptions {
    plan: Plan;
    ws_sender: (o: object) => void;
    updatePanel: (x: number) => void;
    // loadPlan: (x: string) => void;
    // deleteFile: (x: string) => void;
    fileNames: {}
    cancelLoadPlan: () => void;
    setModalInfo: (s: boolean, h: string, n: number, b: object, a: string) => void;
    loadedFileName: string;
    planID: String;
  }


  export interface LoadPanelOptions {
    plan: Plan;
    // ws_sender: (o: object) => void;
    isEdited: boolean;
    updateFileName: (x: boolean) => void;
    updatePanel: (x: number) => void;
    fileNames: {}
    getPlanFromFileName: (fileName: string) => void;
    deleteFile: (file: string) => void;
    loadedFileName: string;
  }

export interface PlanPanelOptions {
    uistatus: { [key: string]: any };
    plan: Plan;
    isEdited: boolean;
    updateFileName: (x: boolean) => void;
    bankAddition: { [key: string]: number };
    fileName: string;
    updatePanel: (x: number) => void;
    layoutSwitch: () => void;
    planOverwrite: (s: object, x: String) => void;
    setModalInfo: (s: boolean, h: string, n: number, b: object, a: string) => void;
    updateSavedFileState: (plan: Plan) => void;
    planID: String;
  }

  export interface PlanSavePanelOptions {
    plan: Plan;
    // ws_sender: (o: object) => void;
    isEdited: boolean;
    updateFileName: (x: boolean) => void;
    planSaved: (f: string, d: object) => void;
    updatePanel: (x: number) => void;
    fileNames: {}
    deleteFile: (file: string) => void;
    setModalInfo: (s: boolean, h: string, n: number, b: object, a: string) => void;
  }

  export interface EditPanelOptions {
    uistatus: {
      [key: string]: any;
    };
    ws_sender: (o: object) => void;
    plan: Plan;
    updatePanel: (x: number) => void;
    planID: String;
  
  }
  export interface CommandPanelOptions {
    uistatus: {
      [key: string]: string;
    };
    ws_sender: (o: object) => void;
    plan: Plan;
    updatePanel: (x: number) => void;
    layoutSwitch: () => void;
    setModalInfo: (s: boolean, h: string, n: number, b: object, a: string) => void;
    loadedFileName: string;
    planID: String;
  }
