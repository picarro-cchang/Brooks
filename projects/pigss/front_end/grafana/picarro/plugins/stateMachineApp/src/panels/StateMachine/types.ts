import { any } from "prop-types";

export interface ButtonInfo {
  caption: string;
  className: string;
  response: string;
}

export interface CommandPanelOptions {
  uistatus: {
    [key: string]: string;
  };
  ws_sender: (o: object) => void;
  plan: Plan;
}

export interface ModalInfo {
  show: boolean;
  html: string;
  num_buttons: number;
  buttons: { [key: string]: ButtonInfo };
}

export interface Plan {
  max_steps: number;
  panel_to_show: number;
  current_step: number;
  focus: PlanFocus;
  last_step: number;
  steps: { [key: string]: PlanStep };
  num_plan_files: number;
  plan_files: { [key: string]: string };
  plan_filename: string;
  bank_names: {
    [key: number]: {
      name: string;
      channels: { [key: number]: string };
    };
  };
}

export interface PlanFocus {
  row: number;
  column: number;
}

export interface PlanLoadPanelOptions {
  plan: Plan;
  ws_sender: (o: object) => void;
  isChanged: boolean;
  updateFileName: (x: boolean) => void;
}

export interface PlanPanelOptions {
  uistatus: { [key: string]: string };
  plan: Plan;
  setFocus: (row: number, column: number) => void;
  ws_sender: (o: object) => void;
  isChanged: boolean;
  updateFileName: (x: boolean) => void;
  // addChanneltoPlan: (bank: number, channel: number) => void;
  bankAddition: {[key: string]: number}
}

export enum PlanPanelTypes {
  NONE = 0,
  PLAN = 1,
  LOAD = 2,
  SAVE = 3,
  EDIT = 4
}

export interface PlanSavePanelOptions {
  plan: Plan;
  ws_sender: (o: object) => void;
  isChanged: boolean;
  updateFileName: (x: boolean) => void;
}

export interface BankConfig {
  chan_mask: number;
  clean: number;
}

export interface PlanStep {
  banks: { [key: number]: BankConfig };
  reference: number;
  duration: number;
}

export interface EditPanelOptions {
  uistatus: {
    [key: string]: any;
  };
  ws_sender: (o: object) => void;
  plan: Plan;
}
