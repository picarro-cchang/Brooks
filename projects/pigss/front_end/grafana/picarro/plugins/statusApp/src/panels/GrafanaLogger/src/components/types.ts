import { GrafanaTheme } from '@grafana/ui';
import { TimeRange } from '@grafana/data';

export interface LogProps {
  level: object[];
  limit: number;
  timeRange: TimeRange;
  data: string[][];
}

export interface LogSectionProps {
  options: LogProps;
  theme: GrafanaTheme;
}

export interface CellProps {
  row: string[];
}
