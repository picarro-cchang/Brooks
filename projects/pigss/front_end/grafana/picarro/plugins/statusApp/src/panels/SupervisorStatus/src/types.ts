import { GrafanaTheme } from '@grafana/ui';

export interface SupervisorStatusProps {
}

export interface SupervisorStatusLayoutProps {
  theme: GrafanaTheme;
  options: SupervisorStatusProps;
}