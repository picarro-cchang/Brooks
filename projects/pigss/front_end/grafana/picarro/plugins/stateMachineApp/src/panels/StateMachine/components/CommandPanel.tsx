import React, { Component, PureComponent } from "react";
import { CommandPanelOptions } from "./../types";

class CommandPanel extends PureComponent<CommandPanelOptions> {
  classNameOpt = {
    DISABLED: "",
    READY: "btn-outline-success",
    ACTIVE: "btn-green"
  };

  getDisabled = (element: string): boolean => {
    let disabled = true;
    const status_dict = this.props.uistatus as any;
    if (element in status_dict) {
      disabled = status_dict[element] === "DISABLED";
    }
    return disabled;
  };

  getClassNameOpt = (element: string): string => {
    let classNames = "";
    const status_dict = this.props.uistatus as any;
    if (element in status_dict) {
      classNames = (this.classNameOpt as any)[status_dict[element]];
    }
    return classNames;
  };

  render() {
    return (
      <div className="panel-command">
        <div style={{ width: "100%", marginTop: 20 }}>
          <div
            className="grid-command"
            style={{
              display: "grid",
              gridTemplateColumns: "1fr 1fr 1fr",
              gridGap: 5,
              padding: 20
            }}
          >
            <button
              onClick={e => this.props.ws_sender({ element: "standby" })}
              disabled={this.getDisabled("standby")}
              value="standby"
              className={
                "btn btn-large btn-command btn-1-3 " +
                this.getClassNameOpt("standby")
              }
            >
              Standby
            </button>
            <button
              onClick={e => this.props.ws_sender({ element: "identify" })}
              disabled={this.getDisabled("identify")}
              value="identify"
              className={
                "btn btn-large btn-command btn-1-3 " +
                this.getClassNameOpt("identify")
              }
            >
              Identify Available Channels
            </button>

            <button
              onClick={e => {
                this.props.ws_sender({ element: "plan" });
              }}
              disabled={this.getDisabled("plan")}
              value="plan"
              className={
                "btn btn-large btn-command btn-1-3 " +
                this.getClassNameOpt("plan")
              }
            >
              Edit Plan
            </button>
            <button
              onClick={e => this.props.ws_sender({ element: "run" })}
              disabled={this.getDisabled("run")}
              value="run"
              className={
                "btn btn-large btn-command btn-run " +
                this.getClassNameOpt("run")
              }
            >
              Run Channel
            </button>
            <button
              onClick={e => this.props.ws_sender({ element: "plan_run" })}
              disabled={this.getDisabled("plan_run")}
              value="plan_run"
              className={
                "btn btn-large btn-command btn-run " +
                this.getClassNameOpt("plan_run")
              }
            >
              Run Plan
            </button>
            <button
              onClick={e => this.props.ws_sender({ element: "plan_loop" })}
              disabled={this.getDisabled("plan_loop")}
              value="plan_loop"
              className={
                "btn btn-large btn-command btn-run " +
                this.getClassNameOpt("plan_loop")
              }
            >
              Loop Plan
            </button>
            <button
              onClick={e => this.props.ws_sender({ element: "reference" })}
              disabled={this.getDisabled("reference")}
              value="reference"
              className={
                "btn btn-large btn-command btn-1-3 " +
                this.getClassNameOpt("reference")
              }
            >
              Reference
            </button>
            <button
              value="edit"
              className={"btn btn-large btn-edit btn-danger"}
              onClick={e => this.props.ws_sender({ element: "edit" })}
              disabled={this.getDisabled("edit")}
            >
              Edit Labels
            </button>
          </div>
        </div>
      </div>
    );
  }
}

export default CommandPanel;
