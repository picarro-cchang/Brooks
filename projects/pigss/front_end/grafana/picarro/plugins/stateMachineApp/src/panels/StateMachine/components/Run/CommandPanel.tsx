import React, { Component, PureComponent } from "react";
import { CommandPanelOptions } from "./../types";

class CommandPanel extends PureComponent<CommandPanelOptions, any> {
  classNameOpt = {
    DISABLED: "",
    READY: "btn-outline-success",
    ACTIVE: "btn-green"
  };

  getDisabled = (element: string): boolean => {
    let disabled = true;
    const status_dict = this.props.uistatus as any;
    if (element in status_dict) {
      disabled = status_dict[element] === "DISABLED";
    }
    return disabled;
  };

  getClassNameOpt = (element: string): string => {
    let classNames = "";
    const status_dict = this.props.uistatus as any;
    if (element in status_dict) {
      classNames = (this.classNameOpt as any)[status_dict[element]];
    }
    return classNames;
  };

  render() {
    console.log("Hello from Command ", this.props.plan)
    return (
      <div className="panel-command">
        <div style={{ width: "100%", marginTop: 20 }}>
          <div
            className="grid-command"
            style={{
              display: "grid",
              gridTemplateColumns: "4rem 4rem 4rem 4rem 4rem 4rem",
              gridGap: 5,
              padding: 20
            }}
          >
            <button
              id={"standby"}
              onClick={e => {
                this.props.ws_sender({ element: "standby" });
              }}
              disabled={this.getDisabled("standby")}
              value="standby"
              className={
                "btn btn-large btn-command btn-1-3 " +
                this.getClassNameOpt("standby")
              }
            >
              Standby
            </button>
            <button
              id={"identify"}
              onClick={e => this.props.ws_sender({ element: "identify" })}
              disabled={this.getDisabled("identify")}
              value="identify"
              className={
                "btn btn-large btn-command btn-1-3 " +
                this.getClassNameOpt("identify")
              }
            >
              Identify Available Channels
            </button>

            <button
              id={"edit-plan"}
              onClick={e => {
                this.props.layoutSwitch();
              }}
              disabled={this.getDisabled("plan")}
              value="plan"
              className={
                "btn btn-large btn-command btn-edit-plan " +
                this.getClassNameOpt("plan")
              }
            >
              Edit Plan
            </button>
            <button
              id={"load-plan"}
              onClick={e => {
                this.props.ws_sender({ element: "load" })
                this.props.updatePanel(1)
              }}
              disabled={this.getDisabled("plan")}
              value="load"
              className={
                "btn btn-large btn-command btn-load " +
                this.getClassNameOpt("plan")
              }
            >
              Load Plan
            </button>
            <button
              id={"run-channel"}
              onClick={e => this.props.ws_sender({ element: "run" })}
              disabled={this.getDisabled("run")}
              value="run"
              className={
                "btn btn-large btn-command btn-run-single " +
                this.getClassNameOpt("run")
              }
            >
              Single Port
            </button>
            <button
              id={"run-plan"}
              onClick={e => {
                this.props.ws_sender({ element: "plan_run" })
                this.props.setModalInfo(true, `<div>Run Plan ${this.props.plan.plan_filename} starting at Step ${this.props.plan.current_step}?</div>`, 3, {
                  1: {
                    caption: "Ok",
                    className: "btn btn-success btn-large",
                    response: {plan: this.props.plan}
                  },
                  2: {
                    caption: "Start at Step 1",
                    className: "btn btn-success btn-large",
                    response: {step: 1, plan: this.props.plan}
                  },
                  3: {
                    caption: "Cancel",
                    className: "btn btn-danger btn-large",
                    response: null
                  }
                }, 'loopPlan')
              }}
              disabled={this.getDisabled("plan_run")}
              value="plan_run"
              className={
                "btn btn-large btn-command btn-run " +
                this.getClassNameOpt("plan_run")
              }
            >
              Run Plan
            </button>
            <button
              id={"loop-plan"}
              onClick={e => {
                console.log("Setting loop plan ", this.state)
                this.props.setModalInfo(true, `<div>Loop Plan ${this.props.plan.plan_filename} starting at Step ${this.props.plan.current_step}?</div>`, 3, {
                  1: {
                    caption: "Ok",
                    className: "btn btn-success btn-large",
                    response: {plan: this.props.plan}
                  },
                  2: {
                    caption: "Start at Step 1",
                    className: "btn btn-success btn-large",
                    response: {step: 1, plan: this.props.plan}
                  },
                  3: {
                    caption: "Cancel",
                    className: "btn btn-danger btn-large",
                    response: null
                  }
                }, 'runPlan')
                this.props.ws_sender({ element: "plan_loop" })
              }}
              disabled={this.getDisabled("plan_loop")}
              value="plan_loop"
              className={
                "btn btn-large btn-command btn-run-loop " +
                this.getClassNameOpt("plan_loop")
              }
            >
              Loop Plan
            </button>
            <button
              id={"reference"}
              onClick={e => this.props.ws_sender({ element: "reference" })}
              disabled={this.getDisabled("reference")}
              value="reference"
              className={
                "btn btn-large btn-command btn-1-3 " +
                this.getClassNameOpt("reference")
              }
            >
              Reference
            </button>
            <button
              id={"edit-labels"}
              value="edit"
              className={"btn btn-large btn-edit btn-danger"}
              onClick={e => this.props.updatePanel(2)}
              disabled={this.getDisabled("edit")}
            >
              Edit Labels
            </button>
          </div>
        </div>
      </div>
    );
  }
}

export default CommandPanel;
