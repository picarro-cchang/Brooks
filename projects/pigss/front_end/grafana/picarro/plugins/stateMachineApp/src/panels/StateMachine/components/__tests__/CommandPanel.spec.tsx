import React from 'react';
import CommandPanel from '../CommandPanel';
import {CommandPanelOptions} from '../../types';
import { shallow, mount } from 'enzyme';
import 'jest-styled-components';
import WS from "jest-websocket-mock";
import { element } from 'prop-types';

const mockClick = jest.fn((element) => {return element;});
const apiLoc = `${window.location.hostname}:8000/controller`;
const socketURL = `ws://${apiLoc}/ws`;

const defaultProps: CommandPanelOptions = {
    uistatus: {
        "standby": "ACTIVE ",
        "identify": "DISABLED",
        "run": "READY",
        "plan": "READY",
        "plan_run": "READY",
        "plan_loop": "READY",
        "reference": "READY",
        "edit": "READY",
    },
    ws_sender: mockClick,
    plan: {
        "max_steps": 32,
        "panel_to_show": 0,
        "current_step": 8,
        "focus": {
            "row": 15,
            "column": 1
        },
        "last_step": 14,
        "steps": {
            "1": {
            "banks": {
                "1": {
                "clean": 0,
                "chan_mask": 1
                },
                "2": {
                "clean": 0,
                "chan_mask": 0
                }
            },
            "reference": 0,
            "duration": 30
            },
            "2": {
            "banks": {
                "1": {
                "clean": 0,
                "chan_mask": 2
                },
                "2": {
                "clean": 0,
                "chan_mask": 0
                }
            },
            "reference": 0,
            "duration": 30
            },
            "3": {
            "banks": {
                "1": {
                "clean": 0,
                "chan_mask": 4
                },
                "2": {
                "clean": 0,
                "chan_mask": 0
                }
            },
            "reference": 0,
            "duration": 30
            },
            "4": {
            "banks": {
                "1": {
                "clean": 0,
                "chan_mask": 8
                },
                "2": {
                "clean": 0,
                "chan_mask": 0
                }
            },
            "reference": 0,
            "duration": 30
            },
            "5": {
            "banks": {
                "1": {
                "clean": 0,
                "chan_mask": 16
                },
                "2": {
                "clean": 0,
                "chan_mask": 0
                }
            },
            "reference": 0,
            "duration": 30
            },
            "6": {
            "banks": {
                "1": {
                "clean": 0,
                "chan_mask": 32
                },
                "2": {
                "clean": 0,
                "chan_mask": 0
                }
            },
            "reference": 0,
            "duration": 30
            },
            "7": {
            "banks": {
                "1": {
                "clean": 0,
                "chan_mask": 64
                },
                "2": {
                "clean": 0,
                "chan_mask": 0
                }
            },
            "reference": 0,
            "duration": 30
            },
            "8": {
            "banks": {
                "1": {
                "clean": 0,
                "chan_mask": 128
                },
                "2": {
                "clean": 0,
                "chan_mask": 0
                }
            },
            "reference": 0,
            "duration": 30
            },
            "9": {
            "banks": {
                "1": {
                "clean": 1,
                "chan_mask": 0
                },
                "2": {
                "clean": 0,
                "chan_mask": 0
                }
            },
            "reference": 0,
            "duration": 30
            },
            "10": {
            "banks": {
                "1": {
                "clean": 0,
                "chan_mask": 0
                },
                "2": {
                "clean": 0,
                "chan_mask": 0
                }
            },
            "reference": 1,
            "duration": 30
            },
            "11": {
            "banks": {
                "1": {
                "clean": 0,
                "chan_mask": 0
                },
                "2": {
                "clean": 0,
                "chan_mask": 1
                }
            },
            "reference": 0,
            "duration": 30
            },
            "12": {
            "banks": {
                "1": {
                "clean": 0,
                "chan_mask": 0
                },
                "2": {
                "clean": 0,
                "chan_mask": 2
                }
            },
            "reference": 0,
            "duration": 30
            },
            "13": {
            "banks": {
                "1": {
                "clean": 0,
                "chan_mask": 0
                },
                "2": {
                "clean": 0,
                "chan_mask": 4
                }
            },
            "reference": 0,
            "duration": 30
            },
            "14": {
            "banks": {
                "1": {
                "clean": 0,
                "chan_mask": 0
                },
                "2": {
                "clean": 0,
                "chan_mask": 8
                }
            },
            "reference": 0,
            "duration": 30
            }
        },
        "num_plan_files": 7,
        "plan_files": {
            "1": "1234567890123456789012345678",
            "2": "30_seconds",
            "3": "__default__",
            "4": "abcdefghijklmnopqrstwxyzabc",
            "5": "abcdefghijklmnopqrstwxyzabcd",
            "6": "kp_test",
            "7": "kp_test_2"
        },
        "plan_filename": "__default__",
        "bank_names": {
            "1": {
            "name": "Bank 1",
            "channels": {
                "1": "Ch. 1",
                "2": "Ch. 2",
                "3": "Ch. 3",
                "4": "Ch. 4",
                "5": "Ch. 5",
                "6": "Ch. 6",
                "7": "Ch. 7",
                "8": "Ch. 8"
            }
            },
            "2": {
            "name": "Bank 2",
            "channels": {
                "1": "Ch. 1",
                "2": "Ch. 2",
                "3": "Ch. 3",
                "4": "Ch. 4",
                "5": "Ch. 5",
                "6": "Ch. 6",
                "7": "Ch. 7",
                "8": "Ch. 8"
            }
            },
            "3": {
            "name": "Bank 3",
            "channels": {
                "1": "Ch. 1",
                "2": "Ch. 2",
                "3": "Ch. 3",
                "4": "Ch. 4",
                "5": "Ch. 5",
                "6": "Ch. 6",
                "7": "Ch. 7",
                "8": "Ch. 8"
            }
            },
            "4": {
            "name": "Bank 4",
            "channels": {
                "1": "Ch. 1",
                "2": "Ch. 2",
                "3": "Ch. 3",
                "4": "Ch. 4",
                "5": "Ch. 5",
                "6": "Ch. 6",
                "7": "Ch. 7",
                "8": "Ch. 8"
            }
            }
        },
    }
};

describe('<CommandPanel />', () => {
    const wrapper = shallow(<CommandPanel {...defaultProps} />);
    const short = mount(<CommandPanel {...defaultProps} />);
    const instance = wrapper.instance() as CommandPanel;
    const server = new WS(socketURL);
    const client = new WebSocket(socketURL);
    
    it('Renders Correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('Buttons Enabled/Disabled Accordingly', () => {
        const value = wrapper.find("button#identify").props().disabled;
        expect(value).toEqual(true);
    });

    it('getDisabled functionality', () => {
        expect(instance.getDisabled("standby")).toEqual(false);
        expect(instance.getDisabled("status")).toEqual(true);
    });

    it('getClassNameOpt functionality', () => {
        expect(instance.getClassNameOpt("status")).toEqual("");
    });

    it('Standby', async () => {

        await server.connected;

        wrapper.find('button#standby').simulate('click');
        const element = mockClick.mock.calls[0][0]['element'];
        
        client.send(element);
        expect(mockClick).toHaveBeenCalled();
        await expect(server).toReceiveMessage("standby");
        expect(mockClick).toHaveReturnedWith({element: "standby"})
        expect(server).toHaveReceivedMessages(["standby"]);
        mockClick.mockClear();
    });
    it('Identify', async () => {

        await server.connected;

        wrapper.find('button#identify').simulate('click');
        const element = mockClick.mock.calls[0][0]['element'];
        
        client.send(element);
        expect(mockClick).toHaveBeenCalled();
        await expect(server).toReceiveMessage("identify");
        expect(mockClick).toHaveReturnedWith({element: "identify"})
        expect(server).toHaveReceivedMessages(["identify"]);
        mockClick.mockClear();
    });

    it('Edit Plan', async () => {

        await server.connected;

        wrapper.find('button#edit-plan').simulate('click');
        const element = mockClick.mock.calls[0][0]['element'];
        
        client.send(element);
        expect(mockClick).toHaveBeenCalled();
        await expect(server).toReceiveMessage("plan");
        expect(mockClick).toHaveReturnedWith({element: "plan"})
        expect(server).toHaveReceivedMessages(["plan"]);
        mockClick.mockClear();
    });

    it('Run Channel', async () => {

        await server.connected;

        wrapper.find('button#run-channel').simulate('click');
        const element = mockClick.mock.calls[0][0]['element'];
        
        client.send(element);
        expect(mockClick).toHaveBeenCalled();
        await expect(server).toReceiveMessage("run");
        expect(mockClick).toHaveReturnedWith({element: "run"})
        expect(server).toHaveReceivedMessages(["run"]);
        mockClick.mockClear();
    });


    it('Run Plan', async () => {

        await server.connected;

        wrapper.find('button#run-plan').simulate('click');
        const element = mockClick.mock.calls[0][0]['element'];
        
        client.send(element);
        expect(mockClick).toHaveBeenCalled();
        await expect(server).toReceiveMessage("plan_run");
        expect(mockClick).toHaveReturnedWith({element: "plan_run"})
        expect(server).toHaveReceivedMessages(["plan_run"]);
        mockClick.mockClear();
    });

    it('Loop Plan', async () => {

        await server.connected;

        wrapper.find('button#loop-plan').simulate('click');
        const element = mockClick.mock.calls[0][0]['element'];
        
        client.send(element);
        expect(mockClick).toHaveBeenCalled();
        await expect(server).toReceiveMessage("plan_loop");
        expect(mockClick).toHaveReturnedWith({element: "plan_loop"})
        expect(server).toHaveReceivedMessages(["plan_loop"]);
        mockClick.mockClear();
    });

    it('Reference', async () => {

        await server.connected;

        wrapper.find('button#reference').simulate('click');
        const element = mockClick.mock.calls[0][0]['element'];
        
        client.send(element);
        expect(mockClick).toHaveBeenCalled();
        await expect(server).toReceiveMessage("reference");
        expect(mockClick).toHaveReturnedWith({element: "reference"})
        expect(server).toHaveReceivedMessages(["reference"]);
        mockClick.mockClear();
    });

    it('Edit Labels', async () => {

        await server.connected;

        wrapper.find('button#edit-labels').simulate('click');
        const element = mockClick.mock.calls[0][0]['element'];
        
        client.send(element);
        expect(mockClick).toHaveBeenCalled();
        await expect(server).toReceiveMessage("edit");
        expect(mockClick).toHaveReturnedWith({element: "edit"})
        expect(server).toHaveReceivedMessages(["edit"]);
        mockClick.mockClear();
    });

});