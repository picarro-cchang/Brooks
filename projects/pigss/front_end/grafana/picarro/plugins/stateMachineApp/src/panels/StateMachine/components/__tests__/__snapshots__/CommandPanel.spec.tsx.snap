// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CommandPanel /> Renders Correctly 1`] = `
<div
  className="panel-command"
>
  <div
    style={
      Object {
        "marginTop": 20,
        "width": "100%",
      }
    }
  >
    <div
      className="grid-command"
      style={
        Object {
          "display": "grid",
          "gridGap": 5,
          "gridTemplateColumns": "1fr 1fr 1fr",
          "padding": 20,
        }
      }
    >
      <button
        className="btn btn-large btn-command btn-1-3 undefined"
        disabled={false}
        id="standby"
        onClick={[Function]}
        value="standby"
      >
        Standby
      </button>
      <button
        className="btn btn-large btn-command btn-1-3 "
        disabled={true}
        id="identify"
        onClick={[Function]}
        value="identify"
      >
        Identify Available Channels
      </button>
      <button
        className="btn btn-large btn-command btn-1-3 btn-outline-success"
        disabled={false}
        id="edit-plan"
        onClick={[Function]}
        value="plan"
      >
        Edit Plan
      </button>
      <button
        className="btn btn-large btn-command btn-run btn-outline-success"
        disabled={false}
        id="run-channel"
        onClick={[Function]}
        value="run"
      >
        Run Channel
      </button>
      <button
        className="btn btn-large btn-command btn-run btn-outline-success"
        disabled={false}
        id="run-plan"
        onClick={[Function]}
        value="plan_run"
      >
        Run Plan
      </button>
      <button
        className="btn btn-large btn-command btn-run btn-outline-success"
        disabled={false}
        id="loop-plan"
        onClick={[Function]}
        value="plan_loop"
      >
        Loop Plan
      </button>
      <button
        className="btn btn-large btn-command btn-1-3 btn-outline-success"
        disabled={false}
        id="reference"
        onClick={[Function]}
        value="reference"
      >
        Reference
      </button>
      <button
        className="btn btn-large btn-edit btn-danger"
        disabled={false}
        id="edit-labels"
        onClick={[Function]}
        value="edit"
      >
        Edit Labels
      </button>
    </div>
  </div>
</div>
`;
